File: backend/routes/contactosRoutes.py
from flask import Blueprint
from controllers.contactosController import (
    get_all_contactos,
    get_contacto_by_id,
    create_contacto,
    update_contacto,
    delete_contacto
)

contactos = Blueprint('contactos', __name__)

@contactos.route('/contactos', methods=['GET'])
def obtener_contactos():
    return get_all_contactos()

@contactos.route('/contactos/<int:id>', methods=['GET'])
def obtener_contacto(id):
    return get_contacto_by_id(id)

@contactos.route('/contactos', methods=['POST'])
def agregar_contacto():
    return create_contacto()

@contactos.route('/contactos/<int:id>', methods=['PUT'])
def modificar_contacto(id):
    return update_contacto(id)

@contactos.route('/contactos/<int:id>', methods=['DELETE'])
def eliminar_contacto(id):
    return delete_contacto(id)

File: backend/routes/contactosRoutes.js
const express = require('express');
const router = express.Router();
const contactosController = require('../controllers/contactosController');

// Obtener todos los contactos
router.get('/', contactosController.getAllContactos);

// Obtener un contacto por ID
router.get('/:id', contactosController.getContactoById);

// Crear un nuevo contacto
router.post('/', contactosController.createContacto);

// Actualizar un contacto existente
router.put('/:id', contactosController.updateContacto);

// Eliminar (soft delete) un contacto
router.delete('/:id', contactosController.deleteContacto);

module.exports = router;
